src/java/org/apache/cassandra/net/MessagingService.java

```diff
@@ -194,7 +194,7 @@
  * implemented in {@link org.apache.cassandra.db.virtual.InternodeInboundTable} and
  * {@link org.apache.cassandra.db.virtual.InternodeOutboundTable} respectively.
  */
-public final class MessagingService extends MessagingServiceMBeanImpl
+public class MessagingService extends MessagingServiceMBeanImpl
```

Comment:
I haven't used that, but for example MigrationCoordinator take MessagingService instance as an argument instead of using the default static instance. The idea was to allow for mocking MessagingService